keanu-project/src/main/java/io/improbable/keanu/util/dot/WriteDot.java

```diff
@@ -0,0 +1,279 @@
+            }
+        }
+        return outputFile;
+    }
+
+
+    /**
+     * Iterates over the vertices of the graph, storing information about vertex labels to display and connections between vertices.
+     *
+     * @param vertex vertex around which the graph will be visualised
+     * @param degree degree of connections to be visualised; for instance, if the degree is 1,
+     *               only connections between the vertex amd it's parents and children will be written out to the DOT file.
+     */
+    private static void obtainGraphInfo(Vertex vertex, int degree) {
+
+        HashSet<VertexId> processedVertices = new HashSet<>();
+        HashSet<Vertex> verticesToProcessNow = new HashSet<>();
+        verticesToProcessNow.add(vertex);
+
+        // Iterate over the graph till the specified degree or till there are no more unprocessed vertices left.
+        int iterationIndex = 0;
+        while(iterationIndex < degree && !verticesToProcessNow.isEmpty()) {
+
+            HashSet<Vertex> verticesToProcessNext = new HashSet<>();
+
+            for (Vertex v : verticesToProcessNow) {
+
+                // Use hashed IDs as the unique identifier for each vertex and store the other information in labels.
+                VertexId vId = v.getId();
+                setLabel(v);
+
+                // Iterate over all vertices connected to this one.
+                Stream<Vertex> connectedVertices = Stream.concat(v.getParents().stream(), v.getChildren().stream());
+                connectedVertices.forEach(connectedVertex -> {
+                    VertexId connectedVId = connectedVertex.getId();
+                    if (!processedVertices.contains(connectedVId)) {
+                        verticeIdsToConnectionInfo.put(concatenateVertexIds(connectedVId, vId), getEdgeString(connectedVId, vId));
+                        verticesToProcessNext.add(connectedVertex);
+                    }
+                });
+
+                // If there is any hyperparameter information to be added to edges, add it now.
+                DotAnnotation vertexAnnotation = v.getClass().getAnnotation(DotAnnotation.class);
+                if (vertexAnnotation != null && vertexAnnotation.displayHyperparameterInfo()) {
+                    applyHyperparameterInfo(v);
+                }
+
+                processedVertices.add(vId);
+            }
+
+            verticesToProcessNow = verticesToProcessNext;
+            iterationIndex++;
+        }
+
+        // Set labels for vertices that have not yet been processed.
+        for (Vertex v : verticesToProcessNow) {
+            setLabel(v);
+        }
+    }
+
+
+    // Utility function for creating a unique identifier for a connection between two vertices.
+    private static String concatenateVertexIds(VertexId v1, VertexId v2) {
+        if (v1.compareTo(v2) < 0) {
+            return v1.toString() + "_" + v2.toString();
+        }
+        else {
```

Comment:
else on same line as '}' I think is our style

Reply:
Oki

Reply:
Is it fine to have it in new line if it is preceded by a comment? Or should I just be more sparing with my comments, as you guys don't seem to be big fans of them?